app/code/Magento/Catalog/Model/Product/Compare/CompareList.php

```diff
@@ -0,0 +1,110 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\Catalog\Model\Product\Compare;
+
+use Magento\Catalog\Model\CompareList as CatalogCompareList;
+use Magento\Catalog\Model\CompareListFactory;
+use Magento\Catalog\Model\ResourceModel\CompareList as CompareListResource;
+
+class CompareList
+{
+    /**
+     * @var CompareListFactory
+     */
+    private $compareListFactory;
+
+    /**
+     * @var CompareListResource
+     */
+    private $compareListResource;
+
+    /**
+     * @param CompareListFactory $compareListFactory
+     * @param CompareListResource $compareListResource
+     */
+    public function __construct(
+        CompareListFactory $compareListFactory,
+        CompareListResource $compareListResource
+    ) {
+        $this->compareListFactory = $compareListFactory;
+        $this->compareListResource = $compareListResource;
+    }
+
+    /**
+     * Get list_id
+     *
+     * @param Item $item
+     *
+     * @return int
+     */
+    public function getListId(Item $item)
```

Comment:
As far as I see, this class has the only one public method for retrieving the list ID. I would suggest renaming the class to `GetItemListId()`.
Also, this class has a new compare list creation as side-effect. I would not recommend hiding it in the service for retrieving list id. This particular class should return either ID or null/exception in case if the list does not exist. 
Then, the parent class responsible for retrieving the list ID should create a new list if there's no exiting list yet. Something like:

```php
$listId = GetItemListId::execute($item);
if ($listId === NULL) {
    $listId = // Create new list 
}

Reply:
File was removed from PR